test/terra/backends/qasm_simulator/qasm_reset.py

```diff
@@ -75,3 +75,15 @@ def test_repeated_resets(self):
             qobj, **self.BACKEND_OPTS).result()
         self.assertSuccess(result)
         self.compare_counts(result, circuits, targets, delta=0)
+
+    def test_reset_moving_qubits(self):
+        """Test QasmSimulator reset with for circuits where qubits have moved"""
+        # count output circuits
+        shots = 1000
+        circuits = ref_reset.reset_circuits_with_entangled_and_moving_qubits(final_measure=True)
+        targets = ref_reset.reset_counts_with_entangled_and_moving_qubits(shots)
+        qobj = assemble(circuits, self.SIMULATOR, shots=shots)
+        result = self.SIMULATOR.run(
+            qobj, **self.BACKEND_OPTS).result()
+        self.assertSuccess(result)
+        self.compare_counts(result, circuits, targets, delta=0.05*shots)
```

Comment:
I'd expect to see two more tests: `initialize` with moving qubits, and `reset` with entangled qubits. I see in the references files that the test of reset has both, so maybe change its name to something like `test_reset_entangled_and_moving_qubits`.

Reply:
You are right that the same tests test both features. I was hesitant about the word `moving` because that aspect is relevant only for mps, and people who run other simulation methods won't understand the name. 
What do you think?

Reply:
Maybe remove the word `moving` from the test name and functions in the reference file; and add a comment somewhere, that we also test moving, which is relevant only for MPS.